File: TicTacToe/TicTacToe_Node/ticTacToe.ts
import { Game } from "./components/game.ts";
import { Player } from "./components/player.ts";
import { PlayingPieceX } from "./components/playingPieces/playingPieceX.ts";
import { PlayingPieceO } from "./components/playingPieces/playingPieceO.ts";

function main() {
  const players: Player[] = [];
  players.push(new Player("Foo", new PlayingPieceX()));
  players.push(new Player("Bar", new PlayingPieceO()));

  const game = new Game(3, players);
  game.startGame();
}

main();

File: TicTacToe/TicTacToe_Go/ticTacToe.go
package main

import (
	"ticTacToe/components"
	"ticTacToe/components/playingPieces"
)

func main() {
	players := make([]*components.Player, 0)
	players = append(players, components.GetNewPlayer("Foo", playingPieces.GetPlayingPieceX()))
	players = append(players, components.GetNewPlayer("Bar", playingPieces.GetPlayingPieceO()))

	game := components.GetNewGame(players, 3)

	game.StartGame()
}
